src/librustc_middle/ty/subst.rs

```diff
@@ -136,6 +136,10 @@ impl<'tcx> GenericArg<'tcx> {
             _ => bug!("expected a const, but found another kind"),
         }
     }
+
+    pub fn is_lifetime(self) -> bool {
+        matches!(self.unpack(), GenericArgKind::Lifetime(..))
+    }
```

Comment:
Not sure I like this, may make it hard to do the right thing, I'd rather see this inlined where it's called.